examples/chart/teleport-kube-agent/templates/statefulset.yaml

```diff
@@ -139,7 +139,7 @@ spec:
           protocol: TCP
         livenessProbe:
           httpGet:
-            path: /readyz
```

Comment:
I think this may have been on purpose as I know our healthz endpoint always just returns 200 so not very useful. @webvictim Do you recall by chance if this was done on purpose?

Reply:
I didn't write the `StatefulSet` version of this chart myself. It's possible that it was an oversight or a misunderstanding of the difference between `/healthz` and `/readyz` - but maybe Kevin had other insight I'm unaware of!

I think the issue as raised is correct because the behaviour should be the same between both `Deployment` and `StatefulSet`.

An interesting thing I learnt recently that `/readyz` won't report healthy after a reconnection until _every_ proxy tunnel has been established - so the time to do this is going to be longer the more proxies you have. I think this should only affect reconnections though rather than the initial startup - I don't know how often Kubernetes runs a readiness check after a successful start.

Reply:
@webvictim readiness probes are only to check if the pod should receive traffic. If it shouldn't receive traffic, then the pod IP is removed from the service/load balancer endpoints list. This monitoring is done throughout the life of the pod and it can be unregistered at any time. Besides that, readiness probes do not cause the pod to restart as opposite to liveness probes.

 In our use case, it is not really important since it's the agent who connects to the proxy and not the opposite